File: Elixir/hello_world_again/lib/k01/k01.ex
defmodule K01 do
  defp question_1 do
    IO.puts("Hello World!　ようこそElixir言語の世界へ！")
  end

  defp question_2 do
    IO.puts("Hello World!")
    IO.puts("ようこそ")
    IO.puts("Elixir言語の世界へ！")
  end

  defp question_3 do
    IO.puts("整数：#{12345}")
    IO.puts("実数：#{123.456789}")
    IO.puts("文字：#{'A'}")
    IO.puts("文字列：#{"ABCdef"}")
  end

  defp question_4 do
    IO.puts("              ##")
    IO.puts("             #  #")
    IO.puts("             #  #")
    IO.puts("            #    #")
    IO.puts("           #      #")
    IO.puts("         ##        ##")
    IO.puts("       ##            ##")
    IO.puts("    ###                ###")
    IO.puts(" ###       ##    ##       ###")
    IO.puts("##        #  #  #  #        ##")
    IO.puts("##         ##    ##         ##")
    IO.puts(" ##     #            #     ##")
    IO.puts("  ###     ##########     ###")
    IO.puts("     ###              ###")
    IO.puts("        ##############")
  end

  @spec execute(integer()) :: :noop | :ok
  def execute(num) do
    case num do
      1 -> question_1()
      2 -> question_2()
      3 -> question_3()
      4 -> question_4()
      _ -> :noop
    end
  end
end

File: Erlang/hello_world/src/questions/k01.erl
-module(k01).

-export([execute/1]).

-spec question_1() -> 'ok'.
question_1() ->
    io:format("Hello World! ようこそErlang言語の世界へ！~n").

-spec question_2() -> 'ok'.
question_2() ->
    io:format("Hello World!~n"),
    io:format("ようこそ~n"),
    io:format("ようこそErlang言語の世界へ！~n").

-spec question_3() -> 'ok'.
question_3() ->
    io:format("整数：~B~n", [12345]),
    io:format("実数：~f~n", [123.456789]),
    io:format("文字：~s~n", ["A"]),
    io:format("文字列：~s~n", ["ABCdef"]).

-spec question_4() -> 'ok'.
question_4() ->
    io:format("~n"),
    io:format("              ##~n"),
    io:format("             #  #~n"),
    io:format("             #  #~n"),
    io:format("            #    #~n"),
    io:format("           #      #~n"),
    io:format("         ##        ##~n"),
    io:format("       ##            ##~n"),
    io:format("    ###                ###~n"),
    io:format(" ###       ##    ##       ###~n"),
    io:format("##        #  #  #  #        ##~n"),
    io:format("##         ##    ##         ##~n"),
    io:format(" ##     #            #     ##~n"),
    io:format("  ###     ##########     ###~n"),
    io:format("     ###              ###~n"),
    io:format("        ##############~n").

-spec execute(_) -> 'ok'.
execute(1) -> question_1();
execute(2) -> question_2();
execute(3) -> question_3();
execute(4) -> question_4();
execute(_) -> 'ok'.
